euro-cordex/cc-plugin-cc6
# Splitting general and CORDEX-Checks, adding further checks

- [x] Create general CV Checker
  - [x] Read CV, MIP tables 
  - [x] Check DRS (compliance with CV and consistency)
  - [x] Check variables, coordinate variables (illegal names) 
  - [x] Check presence of required global attributes
  - [x] Check value of required global attributes
  - [x] Check version_date
  - [x] Check creation_date  
  - [x] Check missing value of main variable 
  - [x] Check time axis, time bounds and `time_range`
  - [x] Optionally writing JSON file with coordinate checksums, global and variable attrs for external consistency checks across files
- [x] Create specific CORDEX-CMIP6 Checker inheriting checks from the general CV Checker
  - [x] Load tables from disk or download CORDEX-CMIP6 MIP tables
  - [x] CORDEX specific checks for file format and compression
  - [x] CORDEX specific check for `time_range`
  - [x] CORDEX specific checks regarding the file chunking
  - [x] CORDEX specific checks for `calendar`
  - [x] CORDEX specific checks for `time:units` 
  - [x] CORDEX specific checks for `driving_variant_label`, `driving_source_id` and `driving_experiment_id`
  - [x] CORDEX specific checks for `domain_id`
  - [x] Check `institution` 
  - [x] CORDEX specific checks for optional/recommended attributes
    - [x] grid `* with * grid spacing *`
    - [x] version_realization_info
    - [x] references
- [x] Test datasets and tests

In further PRs:
- Check coordinates (incl. bounds) and variable
  - variable attributes, coordinate variable attributes
  - vertical coordinates / formula_terms 
  - horizontal coordinates
  - CORDEX-specific: The 2-dimensional geographic latitudes and longitudes of the model grid cells ( lon and lat ) must be also provided as auxiliary coordinates. Longitude coordinates must be strictly monotonically increasing and confined to the range -180 to 360; they must also have absolute values as small as possible given the first two constraints. (e.g., store 170 E to 170 W as 170 to 190, but store 150 W to 130 W as -150 to -130, not 210 to 230.)
  - CORDEX-specific: grid_mapping variable `crs`: `char` or `int`, see [CF-1.11 Appendix F](https://cfconventions.org/Data/cf-conventions/cf-conventions-1.11/cf-conventions.html#appendix-grid-mappings)
- Check domain
- Make use of logger library
- Introduce id for each check
- add tests and (manipulated) test datasets
